src/osyris/spatial/subdomain.py

```diff
@@ -2,11 +2,12 @@
 # Copyright (c) 2022 Osyris contributors (https://github.com/osyris-project/osyris)
 
 import numpy as np
+from .coordinate_transforms import change_origin, change_basis
 from .. import Dataset
 import warnings
 
 
-def extract_sphere(dataset, radius, origin):
+def extract_sphere(dataset, radius, origin, basis=None, dr_L=None):
```

Comment:
I don't think we should add these. I'd prefer if the user did something like `rotate(extract_sphere(dataset, radius, origin), ...)`

Reply:
Why not? I think it's going to be very common for users to want to have their coordinates translated and rotated, and having these as parameters for the subdomain extraction routines shortens their notebooks.

Reply:
From a design point of view, it is always better to start with small and simple building blocks (or Lego pieces) that can be used together, instead of writing functions that take many arguments and can do many things.
Large functions can become very difficult to maintain.

In this case, you are adding the `basis` and `dr_L` to both `extract_sphere` and `extract_box`, and the implementation is duplicated in both functions. If we add further extraction functions in the future, it will be duplicated further.
Then one day, someone will change the arguments to one function and forget to update the others.

We are never preventing users from writing their own wrapper function that would call both.
I would say we start simple, and then wait for feedback from people, and if everyone says that they want to be able to extract and rotate in a single function call, we can make a wrapper function then?

Reply:
You make a very good point, I'll change the method.